### ISSUE TITLE
Better edit experience in table view

### ISSUE THREAD

**Issue by Acylation** [marcusolsson/obsidian-projects#197]
### What would you like help with?

Table displays the items in a plain way, through which we can edit items efficiently. I have some ideas on how to enhance editability of the table view. Generally, an 'Excel/Google sheets like' table view is expected. The projects tab in GitHub is also a good reference. 

- Major inprovement: Accept keyboard input directly and cover the exist value
  - For text variables, directly edit and cover cells' contents by simply typing under 'view mode'
  - For date variables, recon keyboard inputs in several standard formats or YAML-supported format only, then parse and display it.
    - For example, when I type in "2022-11-23" and press enter, the YAML front-matter  is modified and the cell get a date value
    - In case of a non-date input, a criterion should be add right before exit the editing mode to abandon incorrect inputs.
  - For Boolean variables, press the spacebar to switch. (Only a suggestion, there may be a better design)
  - For list/array cells, retype all members separated by comma may be disturbing. I'll mention it below, together with **single selection cells**.
  - Clear a cell's content by pressing 'delete' button
  - Backspace will clear the content and activate the cell

- Major improvement: Better keyboard shortcuts
  - Apply Ctrl+X to date, Boolean, list cells by directly copy YAML string to another file.
  - Set Ctrl+Z  and Ctrl+Y as undo and redo, respectively. (It might require an undo stack across files/notes. I wonder if obsidian provides the API)
  - Set Ctrl+Space as "create new note". If there's a preference on default filename and "default" template, notes can be added **without** modals. It would be quite appreciated if the filenames are also editable.

- Major improvement: Adjust the key 'Enter' and 'Escape'
    - When pressing 'enter' under **view mode**, the cursor moves down rather than switch to the 'edit mode'. (In GitHub projects tab, press enter under view mode will switch to edit mode though, cause there are more types of variables to deal with than excel. I don't prefer this design actually.)
    - When pressing 'enter' under **edit mode**, changes are checked and accepted/abandoned, then the cursor moves down rather than stay on the origin cell. (Excel and GitHub projects tab both apply this)
    - When pressing 'escape' under edit mode, changes are abandoned rather than accepted, and the cursor keeps focusing on the origin cell.

- Minor: Single-select/Multi-select fields support and keyborad enhancement
  - "Priority" or "Status" are common single-select fields. The available key-value list for the whole project can be aquired during loading by traversing through all items. Each item only have one value among them. Once single-select fields be supported, there should be a better input way. A combobox like component is expected, here's how it works.
    - Activate the cell and show the list
      - Activate the cell by a single click on the triangle button
      - Activate the cell by double click (click after setting cursor focus on) on the blank area of the cell
      - Activate the cell by any character typed by keyboard, and input it to filter the options
    - The text in combobox is used to filter the values, and always provide an option called 'create \[new option name\]'
    - Float mouse to navigate the cursor, or switch between options by '↑/up' or '↓/down'. Single click on the option or press 'Enter' to confirm and exit edit mode. 
  - "Tags" or "Assignees" are common multi-select fields. Also, the full available keyvalue list is aquired by traversing through items. However, each file will have multiple options.
  - Activating, typing and navigating act the same as single-select.
  - Scheme1: list with a checkbox
    - Single click on the option or press 'Enter' to add/remove an value
    - Single click outside, or click on a confirm button to accept input and exit edit mode.  
    - <img src="https://user-images.githubusercontent.com/73122375/203571599-679d8dc3-a4da-4962-b271-ca6a8881ddcb.jpg" width="200" />
  - Scheme2: one-by-one insert tags
    - Insert item to the textbox by single click or press 'Enter' on the item, and in textbox they shows like a "tag". List would be closed and the textbox remains. Press enter or click outside/button to confirm
    - To add another value, directly type in a new one, or click the "add" mark next to the tag, which will call out the list again
    - <img src="https://user-images.githubusercontent.com/73122375/203611994-c24fc96d-959f-42e1-94ad-8be512f95279.jpg" width="400" />
    - <img src="https://user-images.githubusercontent.com/73122375/203612005-8f3ab3c3-bfe1-4262-a667-d571d8eae961.jpg" width="400" />
    - Delete item by backspace or click the cross mark on the tag
    - <img src="https://user-images.githubusercontent.com/73122375/203610726-6db8f2e6-12c8-48c7-bd04-dbc576331f58.jpg" width="200" />

**Comment by Acylation** [comment]
If there's any practicable design, I would submit a feature request to track it.

In addition I've just set up the environment and ready to do some coding lol. As I'm new to web development and only have basic cpp knowledge, it may take some times for me to keep up with the rapid iteration, though.

**Comment by marcusolsson** [comment]
Wow! This is amazing feedback! 😍 

My vision for the Table view is to offer a simple but enjoyable experience for everyone, and not just power users. I love that most, if not all, of your suggestions are subtle improvements that will tighten up the experience without bloating the interface with configuration options.

There's a lot of good stuff in here that needs to broken out into separate issues. Let's turn this into an Epic issue where we can discuss the editing experience in the table view, and collect related issues!

Heads-up, I'll have limited availability for the rest of the year, so I don't know when I'll be able to give you more detailed feedback.

(The single/multi select fields sounds related to #76)

**Comment by Acylation** [comment]
Yeah, there are a lot of issues included. I will split them into individual feature requests and find out ways to implement. New issue are added only to describe the feature design more detailed and inspire discussions. Hope that won't bother😉. Some demo versions can be expected if I go through web developments and exist codes smoothly.

> The single/multi select fields sounds related to #76
The enhancement of type system is a vital precondition of these fields. Presently ”list“ field can be used as a model to test the corresponding UI components.

Thanks for your kindly reply and great efforts on the wonderful plugin. Take cares and enjoy the precious holiday time! Also I will "enjoy" the finals ha-ha.
